### ISSUE TITLE
Rename the types

### ISSUE THREAD

**Issue by andrewthad** [andrewthad/unpacked-maybe-numeric#2]
@chessai I started using this in an actual project yesterday, and I realized that the types all being named `Maybe` leads to a bad user experience with haddock when working with data types that have a bunch of these maybe variants as fields. It's difficult to tell what the type of each field actually is. Do you have any strong opinions on renaming to `MaybeInt`, `MaybeInt16`, etc.?

**Comment by chessai** [comment]
No i do not. Also can we use backpack with unboxed types?

**Comment by andrewthad** [comment]
Yes, backpack lets you do levity polymorphic things that you can’t normally do with GHC.  This package would be an excellent candidate for internal use of backpack.  However, there are potential roadblocks: 

1.  cabal build  screws up when trying to install dependencies that use backpack internally.  So, it’s more difficult for people to actually use the library if it uses backpack internally
2.  I’m not sure what the interaction between haddock and modules instantiated via backpack is.

**Comment by chessai** [comment]
How does cabal build screw up?

**Comment by chessai** [comment]
Haddocks im not super worried about, though that does suck. Most of the haddocks will be exactly like Data.Maybe

**Comment by andrewthad** [comment]
To see how cabal screws up, just run

    cabal v2-install sockets
